src/Application.vala

```diff
@@ -246,6 +250,11 @@ public class Terminal.Application : Gtk.Application {
             window.add_tab_with_working_directory (working_directory, null, new_tab);
         }
 
+        // Fallback to a default tab if no tabs were opened 
+        if (window.terminals.length () == 0) {
+            window.add_tab_with_working_directory (Environment.get_home_dir ());
+        }
```

Comment:
this cannot even happen, because we are always creating a tab above.

Reply:
It can happen if the `execute` option is used without any command.